olive/passes/onnx/common.py

```diff
@@ -166,15 +169,20 @@ def model_proto_to_olive_model(
     has_external_data = model_proto_to_file(
         model_proto, output_model_path, **{k: external_data_config[k] for k in config_keys if k in external_data_config}
     )
-    if has_external_data:
+    if has_external_data or external_initializers_name or constant_inputs_name:
         model_path = LocalFolder({"path": Path(output_model_path).parent})
 
         onnx_file_name = Path(output_model_path).name
     else:
         model_path = LocalFile({"path": output_model_path})
         onnx_file_name = None
 
-    olive_model = ONNXModelHandler(model_path=model_path, onnx_file_name=onnx_file_name)
+    olive_model = ONNXModelHandler(
+        model_path=model_path,
+        onnx_file_name=onnx_file_name,
+        external_initializers_name=external_initializers_name,
+        constant_inputs_name=constant_inputs_name,
```

Comment:
is it possible to merge these initializers into the onnx file?

Reply:
It is not possible without manipulating the graph again. The purpose of these files is to have the adapter weights separately so that we can provide different adapter weights to the same base model. 
merging them would make the extraction useless.